examples/gno.land/r/demo/profile/handlers.gno

```diff
@@ -0,0 +1,154 @@
+package profile
+
+import (
+    "bytes"
+    "gno.land/p/demo/ufmt"
+    "gno.land/p/demo/mux"
+    "strings"
+    "std"
+)
+
+const (
+    BaseURL           = "/r/demo/profile"
+    SetStringFieldURL = BaseURL + "?help&__func=SetStringField&field=%s"
+    SetIntFieldURL    = BaseURL + "?help&__func=SetIntField&field=%s"
+    SetBoolFieldURL   = BaseURL + "?help&__func=SetBoolField&field=%s"
+    GetStringFieldURL = BaseURL + "?help&__func=GetStringField&addr=%s"
+    GetIntFieldURL    = BaseURL + "?help&__func=GetIntField&addr=%s"
+    GetBoolFieldURL   = BaseURL + "?help&__func=GetBoolField&addr=%s"
+    ViewAllFieldsURL  = BaseURL + ":u/%s"
+    ViewFieldURL      = BaseURL + ":f/%s/%s"
+)
+
+func homeHandler(res *mux.ResponseWriter, req *mux.Request) {
+    var b bytes.Buffer
+    addresses := make(map[string]struct{})
+
+    b.WriteString("<html><body>")
+    b.WriteString("<h1>Profiles</h1>")
+    b.WriteString("<h2>Setters</h2>")
+    for field := range stringFields {
+        link := ufmt.Sprintf(SetStringFieldURL, field)
+        b.WriteString(ufmt.Sprintf("<p>[<a href='%s'>Set %s Field</a>]</p>", link, field))
+    }
+
+    for field := range intFields {
+        link := ufmt.Sprintf(SetIntFieldURL, field)
+        b.WriteString(ufmt.Sprintf("<p>[<a href='%s'>Set %s Field</a>]</p>", link, field))
+    }
+
+    for field := range boolFields {
+        link := ufmt.Sprintf(SetBoolFieldURL, field)
+        b.WriteString(ufmt.Sprintf("<p>[<a href='%s'>Set %s Field</a>]</p>", link, field))
+    }
+
+    b.WriteString("<p>--------------------------</p>")
+
+    fields.Iterate("", "", func(key string, value interface{}) bool {
```

Comment:
this is consuming a potentially infinite amount of gas which will brick/DOS the contract.